code/game/objects/items/weapons/dice.dm

```diff
@@ -16,17 +16,13 @@
 	icon_state = "d2020"
 	sides = 20
 
-/mob/living/carbon/human/throw_item(atom/target)
-	var/atom/movable/item = src.get_active_hand()
-	if (istype(item, /obj/item/weapon/dice/))
-		var/obj/item/weapon/dice/I = item
-		var/result = rand(1, I.sides)
-		var/comment = ""
-		if(I.sides == 20 && result == 20)
-			comment = "Nat 20!"
-		else if(I.sides == 20 && result == 1)
-			comment = "Ouch, bad luck."
-		I.icon_state = "[I.name][result]"
-		..()
-		usr.visible_message("<span class='notice'>\The [I.name] lands on [result]. [comment]</span>")
-	else ..()
\ No newline at end of file
+/obj/item/weapon/dice/throw_impact(atom/hit_atom)
+	..()
+	var/result = rand(1, sides)
+	var/comment = ""
+	if(sides == 20 && result == 20)
+		comment = "Nat 20!"
+	else if(sides == 20 && result == 1)
+		comment = "Ouch, bad luck."
+	icon_state = "[name][result]"
+	usr.visible_message("<span class='notice'>\The [name] lands on [result]. [comment]</span>")

```

Comment:
Use `src.` (or just leave it unwritten to imply this) instead of `usr.`. `usr.` should be avoided in procs where it is not crystal clear how it's propagated.